packages/ui/src/components/Timeline/Timeline.tsx

```diff
@@ -0,0 +1,108 @@
+import { FC, ReactElement } from 'react'
+import { useMedia } from 'react-use'
+
+import clsx from 'clsx'
+import { compareDesc, format } from 'date-fns'
+import { Check, ChevronRight } from 'lucide-react'
+
+import {
+  EventTimelineConnectorProps,
+  EventTimelineContentProps,
+  EventTimelineIndicatorProps,
+  EventTimelineProps,
+} from './types'
+
+const TimelineIndicator: FC<EventTimelineIndicatorProps> = ({ compareResult, themeColor }): ReactElement | string => {
+  let icon: ReactElement | undefined
+  if (compareResult === -1) {
+    icon = <Check size={10} />
+  } else if (compareResult === 0) {
+    icon = <ChevronRight size={10} />
+  }
+
+  let bgColor = 'white'
+  let borderColor = '#cccccc'
+
+  if (compareResult === -1) {
+    bgColor = themeColor
+    borderColor = themeColor
+  } else if (compareResult === 0) {
+    borderColor = themeColor
+  }
+
+  return (
+    <div
+      className={clsx('flex items-center justify-center  w-4 h-4 rounded-full relative z-10 border')}
+      style={{
+        backgroundColor: bgColor,
+        borderColor: borderColor,
+      }}
+    >
+      {icon}
+    </div>
+  )
+}
+
+const TimelineConnector: FC<EventTimelineConnectorProps> = ({ date, themeColor }) => {
+  const today = format(new Date(), 'yyyy-MM-dd')
+  const eventDate = format(new Date(date), 'yyyy-MM-dd')
+  const compareResult = compareDesc(today, eventDate)
+
+  return (
+    <div className="relative text-center after:inline-block after:absolute after:top-0 after:bottom-0 after:w-0.5 after:bg-gray-200 after:content-[''] after:z-0">
+      <TimelineIndicator compareResult={compareResult} themeColor={themeColor} />
```

Comment:
`compareResult` prop name could be more semantic, like `status`

```ts
  const status = compareResult === -1 ? 'past' : compareResult === 0 ? 'now' : 'future'
```

You could achieve this: